### PR TITLE
moves map code to tmap4 in all chapters

### PR DESCRIPTION
**Pull request by Nowosad**

### CODE REVIEW

**Comment by Nowosad** [comment]
@Robinlovelace Please take a look at it - I've tried to move all of the tmap maps in the book to use tmap v4

**Comment by Robinlovelace** [comment]
Thanks Jakub. Finally taking a look...

File: 02-spatial-data.Rmd [hunk 1]
```diff
@@ -829,39 +833,20 @@ india_buffer_without_s2 = st_buffer(india, 1)
...
 
 tm2 = tm_shape(india_buffer_without_s2) +
-  tm_fill(col = hcl.colors(4, palette = "purple green")[3]) +
+  tm_fill(fill = hcl.colors(4, palette = "purple green")[3]) +
   tm_shape(india) +
-  tm_fill(col = "grey95") +
-  tm_layout(main.title = " ",
-            title = "s2 switched off")
+  tm_fill(fill = "grey95") +
+  tm_title(" ") +
```
Comment (Robinlovelace) [inline 02-spatial-data.Rmd:846]: I like the replacement of `tm_layout(main.title)` with `tm_title()`.

File: 04-spatial-operations.Rmd [hunk 2]
```diff
@@ -58,30 +58,16 @@ canterbury_height = nz_height[canterbury, ]
...
-p_hpnz1 = tm_shape(nz) + tm_polygons(col = "white") +
-  tm_shape(nz_height) + tm_symbols(shape = 2, col = "red", size = 0.25) +
-  tm_layout(main.title = "High points in New Zealand", main.title.size = 1,
-            bg.color = "lightblue")
-p_hpnz2 = tm_shape(nz) + tm_polygons(col = "white") +
+p_hpnz1 = tm_shape(nz) + tm_polygons(fill = "white") +
+  tm_shape(nz_height) + tm_symbols(shape = 2, col = "red", size = 0.5, col_alpha = 0.75) +
+  tm_title("High points in New Zealand") +
+  tm_layout(bg.color = "lightblue")
+p_hpnz2 = tm_shape(nz) + tm_polygons(fill = "white") +
```
Comment (Robinlovelace) [inline 04-spatial-operations.Rmd:65]: I think each of these `tm_*()` calls should be on a new line.

File: 04-spatial-operations.Rmd [hunk 3]
```diff
@@ -764,16 +740,9 @@ The same result can also be generated by piping the output from `st_join()` into
 ```{r spatial-aggregation, echo=FALSE, fig.cap="Average height of the top 101 high points across the regions of New Zealand.", fig.asp=1, message=FALSE, out.width="50%"}
 library(tmap)
 tm_shape(nz_agg) +
-  tm_fill("elevation", breaks = seq(27, 30, by = 0.5) * 1e2) +
+  tm_fill("elevation", 
+          fill.scale = tm_scale(breaks = seq(27, 30, by = 0.5) * 1e2)) +
   tm_borders()
-
-if (packageVersion("tmap") >= "4.0"){
```
Comment (Robinlovelace) [inline 04-spatial-operations.Rmd:770]: Great to be removing these lines!

File: 04-spatial-operations.Rmd [hunk 4]
```diff
@@ -1061,6 +1030,7 @@ The result, shown on the right panel in Figure \@ref(fig:04-ndvi), can be compar
 It allows us to see that the largest NDVI values are connected to areas of dense forest in the northern parts of the area, while the lowest values are related to the lake in the north and snowy mountain ridges.
 
 ```{r 04-ndvi, echo=FALSE, fig.cap="RGB image (left) and NDVI values (right) calculated for the example satellite file of the Zion National Park"}
+# toDo
```
Comment (Robinlovelace) [inline 04-spatial-operations.Rmd:1033]: :+1:

File: figures/lsl-map-1.png
Comment (Robinlovelace) [inline figures/lsl-map-1.png:1]: The new map here is way nicer.

**Review by Robinlovelace — APPROVED** [review]

File: 04-spatial-operations.Rmd (hunk 4 above)
Comment (Nowosad) [reply 04-spatial-operations.Rmd]: Done

**Comment by Nowosad** [comment]
@Robinlovelace I think I fixed all of the issues you've mentioned.

**Comment by Nowosad** [comment]
@Robinlovelace merge it if you think it is okey. (I will do some post-merge cleaning afterwards)

**Comment by Robinlovelace** [comment]
This is awesome Jakub. No more barriers to sending the manuscript I think, let's move on this!

**Pull request merged**